controllers/managedocs_controller.go

```diff
@@ -571,6 +611,11 @@ func (r *ManagedOCSReconciler) reconcilePhases() (reconcile.Result, error) {
 			return ctrl.Result{}, err
 		}
 		if err := r.reconcileEgressNetworkPolicy(); err != nil {
+			if errors.IsNotFound(err) {
+				if err := r.reconcileEgressFirewall(); err != nil {
+					return ctrl.Result{}, err
+				}
+			}
```

Comment:
We do not need to fail logic as we already know which CR to use based on the check in `SetupWithManager`

Reply:
done